src/NewTwitchApi/HelixGuzzleClient.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NewTwitchApi;
+
+use GuzzleHttp\Client;
+
+class HelixGuzzleClient extends Client
+{
+    private const BASE_URI = 'https://api.twitch.tv/helix/';
+
+    public function __construct(string $clientId, array $config = [])
+    {
+        $config = array_merge($config, [
+            'base_uri' => self::BASE_URI,
+            'timeout' => 30,
+            'headers' => ['Client-ID' => $clientId, 'Content-Type' => 'application/json'],
+        ]);
+
+        parent::__construct($config);
```

Comment:
With your current logic, your defaults will overwrite any matching passed in `$config` options.  I'm not sure that's what you want. 

If you want, you could reduce this to
```php
public function __construct(string $clientId, array $config = [])
{
    parent::__construct($config += [
        'base_uri' => self::BASE_URI,
        'timeout' => 30,
        'headers' => ['Client-ID' => $clientId, 'Content-Type' => 'application/json'],
    ]);
}
```

The above assumes that you really want the passed in config to overwrite matching defaults.  If not, swap the arguments of `+=` (for non-numeric keys, `array_merge($a, $b) == $b + $a`)